pkg/yamltemplate/filetests/ytt-library/url-value.yml

```
@@ -0,0 +1,30 @@
+#@ load("@ytt:url", "url")
+
+url_value:
+  #@ testURL1 = url.parse("http://admin:password@url.domain:8080")
+  #@ testURL2 = testURL1.without_user()
+  #@ testURL3 = url.parse("http://admin@url.domain:8080")
+  test1: #@ testURL1.string()
+  test2: #@ testURL1.user
+  test3: #@ testURL1.user.name
+  test4: #@ testURL1.user.password
+  test5: #@ testURL2.string()
+  test6: #@ testURL2.user
+  test7: #@ testURL3.string()
+  test8: #@ testURL3.user
+  test9: #@ testURL3.user.name
+  test10: #@ testURL3.user.password
```

```suggestion
  test1: #@ testURL1.string()
  test2: #@ testURL1.user.string()
  test3: #@ testURL1.user.name
  test4: #@ testURL1.user.password
  test5: #@ testURL2.string()
  test6: #@ testURL2.user.string()
  test7: #@ testURL3.string()
  test8: #@ testURL3.user.string()
  test9: #@ testURL3.user.name
  test10: #@ testURL3.user.password
```

I created an inconsistency, here. The rationale that we came to for wanting to require that `urlValue` to no assume an encoding... well, not sure why it didn't occur to me that the same is true for `urlValue.user` 😬 .

---

Small correction here, in `test6: #@ testURL2.user.string()`, we actually should just keep it `test6: #@ testURL2.user` as `testURL2` doesn't contain user information, so its `user` member will be a `None` object.